Makes/ViewController.m

```diff
@@ -0,0 +1,146 @@
+    
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"url_string_file" ofType:@"txt"];
+    NSString *content = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:nil];
+
+    
+    [manager GET:content parameters:nil progress:nil success:^(NSURLSessionTask *task, id responseObject) {
+        
+        NSArray* dictionariesArray = (NSArray*)responseObject;
+        NSLog(@"%@",responseObject);
+        
+        Make* makeHolder = [[Make alloc]init];
+        NSError * parserError;
+        NSMutableArray* indexPaths = [NSMutableArray array];
+        
+        for(int i=0; i<dictionariesArray.count; i++)
+        {
+            makeHolder = [MTLJSONAdapter modelOfClass:[Make class] fromJSONDictionary:[dictionariesArray objectAtIndex:i] error:&parserError];
+            
+            if(parserError)
+            {
+                NSLog(@"%@",parserError);
+            }
+            
+            [self.makes addObject:makeHolder];
+            NSLog(@"%@",[(Make*)[self.makes objectAtIndex:i] nameAr]);
+            [indexPaths addObject:[NSIndexPath indexPathForItem:self.makes.count-1 inSection:0]];
+        }
+        
+        [self.makesCollectionView insertItemsAtIndexPaths:indexPaths];
+        
+    } failure:^(NSURLSessionTask *operation, NSError *error) {
+        NSLog(@"Error: %@", error);
+    }];
+
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    self.makes = [[NSMutableArray alloc]init];
+    self.imagesCache = [[NSCache alloc]init];
+    self.title = @"MAKES";
+    self.view.backgroundColor = [UIColor lightGrayColor];
+    
+    [self loadMakes];
+    
+    UICollectionViewFlowLayout *layout=[[UICollectionViewFlowLayout alloc] init];
+    self.makesCollectionView =[[UICollectionView alloc] initWithFrame:self.view.frame collectionViewLayout:layout];
+    [self.makesCollectionView setDataSource:self];
+    [self.makesCollectionView setDelegate:self];
+    
+    [self.makesCollectionView setBackgroundColor:[UIColor lightGrayColor]];
+    [self.makesCollectionView registerClass:[MakesCollectionViewCell class] forCellWithReuseIdentifier:@"cellIdentifier"];
+    [self.view addSubview:self.makesCollectionView];
+    
+    
+    UIEdgeInsets padding = UIEdgeInsetsMake(10 , 10, 10, 10);
+    
+    [self.makesCollectionView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.equalTo(self.view).with.insets(padding);
+    }];
+    
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
+{
+    return self.makes.count;
+}
+
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    
+    MakesCollectionViewCell *cell=(MakesCollectionViewCell*)[collectionView dequeueReusableCellWithReuseIdentifier:@"cellIdentifier" forIndexPath:indexPath];
+    
```

Comment:
It should be like this  MakesCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cellIdentifier"  forIndexPath:indexPath] there's no need to casting it as it returns makes collection view cel already